pandas/plotting/__init__.py

```diff
@@ -4,12 +4,6 @@
 
 # flake8: noqa
 
-try:  # mpl optional
-    from pandas.plotting import _converter
-    _converter.register()  # needs to override so set_xlim works with str/number
```

Comment:
One problem with this is *I think* this will affect users who just use matplotlib for plotting but have pandas imported (now they will get nicer datetime formatting)

Not sure this is a reason not to do this, though, as exactly this side-effect also has come up before as a problem. But it would change the moment when the side-effect is activated (on pandas import or on first pandas plot)

Reply:
Yep. I was about to make a PR to update their documentation. I think this is unavoidable.

Reply:
I *guess* we could make an option, and if users really wanted they could set something in their ipython config. But I'm tempted to wait.

Reply:
In https://github.com/pandas-dev/pandas/issues/16764#issuecomment-310694497 you had a patch with a global registered flag. Not sure how important it is to not again register on each plotting call.

In eg a case where a user manually removed entries from the registry to not have pandas things there, this could help.

Reply:
I still have that flag https://github.com/pandas-dev/pandas/pull/17710/files#diff-2b118bda866f3d626ceb6b529c62cd1aR46

Is the desired behavior that we should only attempt to register once? So if a user removes the pandas converters, they don't get re-added on each call? That's what happens currently.

Reply:
Ah, should have overlooked that in the diff.

Do we need to expose a user-oriented function to register the converters? (eg `pandas.plotting.register_converters()`)

Reply:
Ah, I see that in the matplotlib docs they already use the one through pandas.tseries.converter (https://github.com/matplotlib/matplotlib/pull/9251/files) (although the location has changed)